src/Symfony/Component/Notifier/Bridge/MicrosoftTeams/MicrosoftTeamsTransport.php

```diff
@@ -56,12 +57,20 @@ protected function doSend(MessageInterface $message): SentMessage
             throw new UnsupportedMessageTypeException(__CLASS__, ChatMessage::class, $message);
         }
 
+        if ($message->getOptions() && !$message->getOptions() instanceof MicrosoftTeamsOptions) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" for options.', __CLASS__, MicrosoftTeamsOptions::class));
+        }
+
+        $options = ($opts = $message->getOptions()) ? $opts->toArray() : [];
+
+        if (!$options) {
```

Comment:
Is this correct? We set `title` only if $options is empty?

Or should we set thet title if no title exists?

Reply:
We have the following tests which are passing:
```php
    public function testSend()
    {
        $message = 'testMessage';

        $expectedBody = json_encode(['title' => $message]);

        $client = new MockHttpClient(function (string $method, string $url, array $options = []) use ($expectedBody): ResponseInterface {
            $this->assertJsonStringEqualsJsonString($expectedBody, $options['body']);

            return new MockResponse('1', ['response_headers' => ['request-id' => ['testRequestId']], 'http_code' => 200]);
        });

        $transport = $this->createTransport($client);

        $transport->send(new ChatMessage($message));
    }

    public function testSendWithNotification()
    {
        $notification = new Notification('testMessage');
        $chatMessage = ChatMessage::fromNotification($notification);
        $options = MicrosoftTeamsOptions::fromNotification($notification);

        $expectedBody = json_encode([
            'title' => $options->toArray()['title'],
        ]);

        $client = new MockHttpClient(function (string $method, string $url, array $options = []) use ($expectedBody): ResponseInterface {
            $this->assertJsonStringEqualsJsonString($expectedBody, $options['body']);

            return new MockResponse('1', ['response_headers' => ['request-id' => ['testRequestId']], 'http_code' => 200]);
        });

        $transport = $this->createTransport($client);

        $transport->send($chatMessage);
    }
```

🤔

Reply:
Good. I like https://github.com/symfony/symfony/pull/40738/commits/4a8930242cdd40105638a6ee3af93eb1023eab82